docs/designs/reference_link.md

```diff
@@ -0,0 +1,226 @@
+[!code-csharp[Main](subfolder/program.cs)]
+```
+
+All links in `file2.md` are relative to the `file2.md` itself, even when it's included by `file1.md`.
+
+> [!Note]
+> Please note that the file path in include syntax is handled differently than Markdown link.
+> You can only use **relative path** to specify location of the included file.
+> And DocFX doesn't require included file to be included in `docfx.yml`.
+>
+> [!Tip]
+> Each file in `docfx.yml` will build into an output file. But included files usually don't need to build into individual
+> topics. So it's not recommended to include them in `docfx.yml`, they should be excluded from the initial scope `docfx.yml` if needed.
+
+### Include a [nested toc](table-of-contents.md#link-to-another-toc-file) using relative path
+
+The [toc](table-of-contents.md) syntax support to reference a nested toc using relative path or relative folder.
+
+For example, if `toc.md` reference `subfolder\toc.md`:
+
+```markdown
+#[child](subfolder\toc.md)
+```
+
+or 
+
+```markdown
+#[child](subfolder\)
+```
+
+All links in `subfolder\toc.md` are relative to the `subfolder\toc.md` itself, even when it's included by `toc.md`.
+
+### The other ways to link to a local resource
+
+#### Relative path starts with `~`
+
+Sometimes you may find it's complicated to calculate relative path between two files.
+DocFX also supports path starts with `~` to represent path relative to the root directory of your project (i.e. where `docfx.yml` is located).
+This kind of path will also be validated and resolved during build.
+
+For example, you can write the following links in `subfolder\file2.md` to reference `file1.md`:
+ 
+```markdown
+[file1](~/file1.md)
+
+[file1](../file1.md)
+```
+
+Both will be resolved to `../file1.html`.
+
+#### Absolute path starts with `\` or `/`
+
+It's recommended to always use relative path to reference another file in the same project. Relative path will be resolved during build and produce build warning if the target file does not exist.
+
+You can also use absolute path (path starts with `/`) to link to another file, but DocFX won't check its correctness for you and will keep it **as-is** in the output HTML.
+That means you should use the output file path as absolute path. For example, in the above case, you can also write the link as follows:
+
+```markdown
+[file2](/subfolder/file2.html)
+```
+
+#### Relative path in [inline HTML](https://daringfireball.net/projects/markdown/syntax#html)
+
+DocFX also supports to use relative path in inline HTML. Path in HTML link (`<a>`), image (`<img>`), script (`<script>`) and css (`<link>`) will also be resolved if they're relative path
+
+> [!Warning]
+> [Automatic link](https://daringfireball.net/projects/markdown/syntax#autolink) doesn't support relative path.
+> If you write something like `<file.md>`, it will be treated as an HTML tag rather than a link.
+
+## Link to a resource stored in [dependent repository](config.md)
```

Comment:
give an anchor

Reply:
it's not clearly defined in config.md yet, so i added a TODO at the top of this doc

---
In reply to: [187242924](https://github.com/dotnet/docfx/pull/2752#discussion_r187242924) [](ancestors = 187242924)